.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,131 @@
+#   3. Repository variables: SONAR_ORG_KEY, SONAR_PROJECT_KEY (Project → Information).
+#
+# Until all three are set, this workflow skips with a notice (does not fail CI).
+#
+# Optional Telegram on Sonar job failure only (TELEGRAM_BOT_TOKEN + TELEGRAM_CHAT_ID).
+
+name: SonarCloud
+
+on:
+  push:
+    branches:
+      - ${{ vars.PRIMARY_BRANCH || 'main' }}
```

actionlint (1.7.12)

[error] character ' ' is invalid for branch and tag names. ref name cannot contain spaces, ~, ^, :, [, ?, *. see `man git-check-ref-format` for more details. note that regular expression is unavailable. note: filter pattern syntax is explained at https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

(glob)